src/Kernels/KernelTensors.jl

```diff
@@ -38,22 +40,20 @@ Represents a Collision-Coalescence kernel.
 # Fields
 
 """
-struct CoalescenceTensor{FT} <: KernelTensor{FT}
-    "polynomial order of the tensor"
-    r::Int
+struct CoalescenceTensor{N, FT, M} <: KernelTensor{FT}
     "collision-coalesence rate matrix"
-    c::Array{FT}
+    c::SMatrix{N, N, FT, M}
 
-    function CoalescenceTensor(c::Array{FT}) where {FT <: Real}
+    function CoalescenceTensor(c::SMatrix{N, N, FT, M}) where {N, M, FT <: Real}
         check_symmetry(c)
-        new{FT}(size(c)[1] - 1, c)
+        new{N, FT, M}(c)
     end
 end
 
 
-function CoalescenceTensor(kernel_func, order::Int, limit::FT; lower_limit::FT = FT(0)) where {FT <: Real}
+function CoalescenceTensor(kernel_func, order::Int, limit::FT; lower_limit::FT = FT(0)) where {FT <: Real} # TODO: kwarg?
```

Comment:
Do you want to keep these TODOs in the code that's merged in? (it's fine if you do, just wanted to point it out :) )